### PR TITLE
LPD-33375 | master

### PR DESCRIPTION
**Pull request by marcosapmf**
Jira ticket: https://liferay.atlassian.net/browse/LPD-33375

### CODE REVIEW

**Comment by marcosapmf** [comment]
ci:test:analytics-cloud

**Comment by marcosapmf** [comment]
ci:test:sf

**Comment by marcosapmf** [comment]
ci:test:sf

**Comment by marcosapmf** [comment]
ci:test:analytics-cloud

File: modules/test/playwright/tests/osb-faro-web/utils/navigation.ts [hunk 1]
```diff
@@ -49,3 +49,17 @@ export async function navigateToACSitesPageViaURL({
...
+	projectID,
+}: {
+	channelID: string;
+	page: Page;
+	projectID: string;
+}) {
+	await page.goto(
+		`${faroConfig.environment.baseUrl}/workspace/${projectID}/${channelID}/tests`
+	);
+}
```
Comment (ViniLopes20) [inline modules/test/playwright/tests/osb-faro-web/utils/navigation.ts:65]: Hey @marcosapmf, I think creating a function to navigate to each page URL in AC won't be very good. I think we'll have several similar functions just for navigation. I think we could follow one of these suggestions:

1. Create an `enum` with the final term of the URL that represents each page, then we could use the `enum` values ​​in a single function and the function would receive as a parameter the value of the `enum` that would be used:
```
export enum ACPages {
	sitePage = 'sites',
	assetPage = 'assets',
	eventAnalysisPage = 'events',
	segmentPage = 'segments',
	individualPage = 'individuals',
	testPage = 'tests',
}

export async function navigateToACPageViaURL({
	ACPage,
	channelID,
	page,
	projectID,
}: {
	ACPage: ACPages;
	channelID: string;
	page: Page;
	projectID: string;
}) {
	await page.goto(
		`${faroConfig.environment.baseUrl}/workspace/${projectID}/${channelID}/${ACPage}`
	);
}
```

2. We can just use `navigateToACSitesPageViaURL` to access AC via URL and then navigate in the UI to the pages we want to access in sequence using the `navigateTo` function.

What do you think?

File: modules/test/playwright/tests/osb-faro-web/ab-test.spec.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+		const channelName = 'My Property - ' + getRandomString();
+
+		const {channel, project} = await syncAnalyticsCloud({
+			apiHelpers,
+			channelName,
+			page,
+			siteName,
+		});
+
+		await test.step('Go to AC Page', async () => {
```
Comment (ViniLopes20) [inline modules/test/playwright/tests/osb-faro-web/ab-test.spec.ts:67]: Hey @marcosapmf, Update this description

File: modules/test/playwright/tests/osb-faro-web/ab-test.spec.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+			await navigateToSitePage({
+				page,
+				pageName: pageTitle,
+				siteName,
+			});
+			await page.waitForTimeout(10000);
```
Comment (ViniLopes20) [inline modules/test/playwright/tests/osb-faro-web/ab-test.spec.ts:73]: Hey @marcosapmf, I think this `waitForTimeout` is not necessary. Could you try to put a `waitForSelector` to wait for the AB Test icon to appear on the screen? What do you think?

File: modules/test/playwright/tests/osb-faro-web/utils/ab-test.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+	await createVariantButton.click();
+
+	const variantModal = await page.locator('.modal-content');
+
+	await variantModal.locator('input').fill(name);
+
+	await variantModal.getByText('Save').click();
+
+	await expect(page.locator(`[data-title="${name}"]`)).toBeVisible();
+}
```
Comment (ViniLopes20) [inline modules/test/playwright/tests/osb-faro-web/utils/ab-test.ts:48]: Hey @marcosapmf, one question for you. These functions will interact on the DXP side. Is it not necessary to create a module and file related to the respective module that we have to represent the AB test on the DXP side?

File: modules/test/playwright/tests/osb-faro-web/utils/navigation.ts (hunk 1 above)
Comment (marcosapmf) [reply modules/test/playwright/tests/osb-faro-web/utils/navigation.ts:65]: Agreed @ViniLopes20. Let's follow your first suggestion :)

**Comment by marcosapmf** [comment]
ci:test:analytics-cloud

**Comment by marcosapmf** [comment]
ci:test:sf

**Comment by marcosapmf** [comment]
ci:test:analytics-cloud

File: modules/test/playwright/tests/osb-faro-web/utils/navigation.ts [hunk 5]
```diff
@@ -8,6 +8,15 @@ import {Page} from '@playwright/test';
...
 
+export enum ACPage {
+	sitePage = 'sites',
+	assetPage = 'assets',
+	eventAnalysisPage = 'events',
+	segmentPage = 'segments',
+	individualPage = 'individuals',
+	testPage = 'tests',
+}
+
```
Comment (ViniLopes20) [inline modules/test/playwright/tests/osb-faro-web/utils/navigation.ts:19]: Hey @marcosapmf, I think I put the URL term for the individuals and segments pages wrong in the other comment. Here is the correct one

```suggestion
export enum ACPage {
	sitePage = 'sites',
	assetPage = 'assets',
	eventAnalysisPage = 'event-analysis',
	segmentPage = 'contacts/segments',
	individualPage = 'contacts/individuals',
	testPage = 'tests',
}

```

File: modules/test/playwright/tests/osb-faro-web/segments.spec.ts [hunk 6]
```diff
@@ -303,7 +305,8 @@ test(
 		});
 
 		await test.step('Go to Analytics Cloud and Switch the property', async () => {
-			await navigateToACSitesPageViaURL({
+			await navigateToACPageViaURL({
+				acPage: ACPage.sitePage,
```
Comment (ViniLopes20) [inline modules/test/playwright/tests/osb-faro-web/segments.spec.ts:309]: Hey @marcosapmf, I think there should be a navigation to the segments page and you could remove line 317:
`await navigateTo({page, pageName: 'Segments'});`

```suggestion
				acPage: ACPage.segmentPage,
```

Comment (ViniLopes20) [reply modules/test/playwright/tests/osb-faro-web/segments.spec.ts:309]: Same comment for all occurrences of `navigateToACPageViaURL` after this point

File: modules/test/playwright/tests/osb-faro-web/page.spec.ts [hunk 7]
```diff
@@ -153,7 +154,8 @@ test(
 		});
 
 		await test.step('Go to Analytics Cloud and Switch the property', async () => {
-			await navigateToACSitesPageViaURL({
+			await navigateToACPageViaURL({
+				acPage: ACPage.sitePage,
```
Comment (ViniLopes20) [inline modules/test/playwright/tests/osb-faro-web/page.spec.ts:158]: Hey @marcosapmf, I think there should be a navigation to the segments page and you could remove line 166:
await navigateTo({page, pageName: 'Segments'});

```suggestion
			await navigateToACPageViaURL({
				acPage: ACPage.segmentPage,
```

File: modules/test/playwright/tests/osb-faro-web/individuals.spec.ts [hunk 8]
```diff
@@ -79,7 +79,8 @@ test(
 		});
 
 		await test.step('Go to Analytics Cloud and Switch the property', async () => {
-			await navigateToACSitesPageViaURL({
+			await navigateToACPageViaURL({
+				acPage: ACPage.sitePage,
```
Comment (ViniLopes20) [inline modules/test/playwright/tests/osb-faro-web/individuals.spec.ts:83]: Hey @marcosapmf, I think there should be a navigation to the Individuals page and you could remove line 91:
`await navigateTo({page, pageName: 'Individuals'});`

```suggestion
			await navigateToACPageViaURL({
				acPage: ACPage.individualPage,
```

**Review by ViniLopes20 — CHANGES_REQUESTED** [review]
Very Good, @marcosapmf 🎉. Thanks for working on this. I just put some suggestions.

PS: I think your test is failing in CI because the emulator is breaking. Try running in CI with only the Playwright tests

**Comment by marcosapmf** [comment]
ci:test:analytics-cloud

**Comment by marcosapmf** [comment]
ci:test:analytics-cloud

**Comment by marcosapmf** [comment]
ci:test:sf

**Comment by marcosapmf** [comment]
ci:test:analytics-cloud

**Comment by ViniLopes20** [comment]
ci:test:analytics-cloud

**Review by ViniLopes20 — APPROVED** [review]
Very good @marcosapmf 🎉

**Comment by ViniLopes20** [comment]
Hey @marcosapmf, you need to delete the `ABTestACSide.testcase` file because this file currently has no tests and is causing issues when we try to run the tests on CI

**Comment by marcosapmf** [comment]
ci:test:sf

**Comment by ViniLopes20** [comment]
ci:test:analytics-cloud

**Comment by marcosapmf** [comment]
ci:test:sf

**Comment by marcosapmf** [comment]
resent https://github.com/liferay-ac/liferay-portal/pull/1741

**Pull request closed without merging**